routetable/route_table.go

```diff
@@ -329,6 +376,35 @@ func (r *RouteTable) Apply() error {
 	return nil
 }
 
+func (r *RouteTable) ensureL2Dataplane(linkAttrs *netlink.LinkAttrs, target L2Target) error {
+	// For each L2 entry we need to program, program it.
+	// Add a static ARP entry.
+	a := &netlink.Neigh{
+		LinkIndex:    linkAttrs.Index,
+		State:        netlink.NUD_PERMANENT,
+		Type:         syscall.RTN_UNICAST,
+		IP:           target.GW.AsNetIP(),
+		HardwareAddr: target.VTEPMAC,
+	}
+	if err := netlink.NeighSet(a); err != nil {
```

Comment:
It seems to be from the description, but not sure if a "replace" to the same thing that's currently programmed is disruptive to the data plane or not.

I'd assume not, but a quick search doesn't show any obvious results. This is the same operation that flannel does, but flannel won't do this periodically IIUC so it might not be relevant there.

Reply:
Running an `ip monitor`, I can see that the current code does write new neighbor entries periodically, but I don't see any deletes, so I suspect that we're all good here in terms of data plane.